### ISSUE TITLE
Blobiness measures

### ISSUE THREAD

**Issue by Hrovatin** [theislab/cross_system_integration#7]
Spatial descriptive statistics: may be more appropriate for points within a pre-defined region. https://en.wikipedia.org/wiki/Spatial_descriptive_statistics
Some adaptation of LOF: https://en.wikipedia.org/wiki/Local_outlier_factor
KDE, but performance degrades in higher dim: https://scikit-learn.org/stable/modules/density.html#density-estimation Also may be hard to determine the bin width especially if different dim different range - thus UMAP in n dim may be useful. Gaussian may not be good as oversmooth https://jax.readthedocs.io/en/latest/_autosummary/jax.scipy.stats.gaussian_kde.html
maybe the above could be to some extent approximated by distn of distance to K nearest neighbours - but that likely wont capture disconnected communities - only less/more dense regions in places where points are present. The KDE also wont capture that if do filtering of low density regions - so anyway need sth for disjoint graph components.
